compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs

```diff
@@ -79,9 +77,9 @@ impl<'tcx> LateLintPass<'tcx> for OpaqueHiddenInferredBound {
         for (pred, pred_span) in
             cx.tcx.explicit_item_bounds(def_id).instantiate_identity_iter_copied()
         {
-            // Liberate bound regions in the predicate since we
-            // don't actually care about lifetimes in this check.
-            let predicate = cx.tcx.liberate_late_bound_regions(def_id, pred.kind());
+            let predicate = infcx.instantiate_binder_with_placeholders(
```

Comment:
Not entirely sure if we should instantiate with fresh vars instead :sweat_smile:

Reply:
placeholders i guess are fine since we don't solve regions at all